segmentation/include/pcl/segmentation/euclidean_cluster_comparator.h

```diff
@@ -162,22 +124,21 @@ namespace pcl
         exclude_labels_ = boost::make_shared<std::vector<bool> >(exclude_labels);
       }
 
-      /** \brief Compare points at two indices by their plane equations.  True if the angle between the normals is less than the angular threshold,
-        * and the difference between the d component of the normals is less than distance threshold, else false
+      /** \brief Compare points at two indices by their euclidean distance
         * \param idx1 The first index for the comparison
         * \param idx2 The second index for the comparison
         */
       virtual bool
       compare (int idx1, int idx2) const
       {
-        int label1 = labels_->points[idx1].label;
-        int label2 = labels_->points[idx2].label;
-        
-        if (label1 == -1 || label2 == -1)
-          return false;
-        
-        if ( (*exclude_labels_)[label1] || (*exclude_labels_)[label2])
-          return false;
+        if (labels_ && exclude_labels_)
+        {
+          assert (labels_->size () == input_->size ());
+          const uint32_t &label1 = labels_->at (idx1).label;
+          const uint32_t &label2 = labels_->at (idx2).label;
+          if ( (*exclude_labels_)[label1] || (*exclude_labels_)[label2])
```

Comment:
Is it really necessary? Both labels_ and exclude_labels_ are set by the user... I would personally just let it explode and have the user debug it.